### ISSUE TITLE
space-before-function-paren and async function

### ISSUE THREAD

**Issue by jupl** [babel/babel-eslint#168]
The following line:

```js
it('handles updates', async function() {
                           ^
```

Is throwing up a flag under the caret with the violation of `space-before-function-paren`. This occurs in v4.0.7 and has not thrown an error in previous versions.

**Comment by hzoo** [comment]
Thanks for the report - I'l try to fix this soon. It's definitely because of https://github.com/babel/babel-eslint/pull/166. Need to fix this and the other issue somehow - but this is a good test to add in.

**Comment by hzoo** [comment]
Ok should be fixed in `4.0.8`

**Comment by taion** [comment]
I'm still seeing issues here against `v4.0.10`. For:

```js
async function foo() {
  return 5;
}

export default foo;
```

I get

```
test.js
  1:15  error  Unexpected space before function parentheses  space-before-function-paren
```

**Comment by hzoo** [comment]
Dang - but hmm cannot reproduce? on eslint@1.1.0 as well.

```
//.eslintrc
parser: "babel-eslint"
rules:
    space-before-function-paren: [2, "never"]
```

**Comment by taion** [comment]
Sorry - something terribly messed up in my environment. It went away.
